test/views/supplementary/PreviousDocumentsViewSpec.scala

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.supplementary
+import forms.supplementary.Document
+import helpers.views.supplementary.{CommonMessages, PreviousDocumentsMessages}
+import play.api.data.Form
+import play.twirl.api.Html
+import views.html.supplementary.previous_documents
+import views.supplementary.spec.ViewSpec
+
+class PreviousDocumentsViewSpec extends ViewSpec with PreviousDocumentsMessages with CommonMessages {
+
+  private val form: Form[Document] = Document.form()
+  private def createView(form: Form[Document] = form): Html = previous_documents(form, Seq())(fakeRequest, messages, appConfig)
+
+  "Previous Documents View" should {
+
+    "have proper messages for labels" in {
+
+      assertMessage(previousDocuments, "Previous documents")
+      assertMessage(title, "2/1 Enter previous DUCR or MUCR references linked to these goods")
+      assertMessage(hint, "For example, 8GB-123456789101-SHIP1")
+      assertMessage(documentX, "Temporary storage")
+      assertMessage(documentY, "Simplified declaration")
+      assertMessage(documentZ, "Previous document")
+      assertMessage(documentType, "Enter previous document code")
+      assertMessage(documentReference, "Enter the reference for the previous DUCR or MUCR")
+      assertMessage(documentGoodsIdentifier, "Enter the goods item this previous document relates to")
+      assertMessage(documentCategoryLabel, "Document Category")
+      assertMessage(documentTypeLabel, "Document Type")
+      assertMessage(documentReferenceLabel, "Document Reference")
+      assertMessage(documentGoodsIdentifierLabel, "Goods Item Identifier")
+      assertMessage(removePackageInformation, "Remove Packaging Information")
+    }
+
+    "have proper messages for error labels" in {
+
+      assertMessage(documentCategoryEmpty, "Document category cannot be empty")
```

Comment:
code is better organised and readability improves, probably maintenance improves as well.
I will do that as part of my new story. ;)